packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -188,10 +188,10 @@ export class KubernetesClient {
     const kubernetesConfiguration = this.configurationRegistry.getConfiguration('kubernetes');
     const userKubeconfigPath = kubernetesConfiguration.get<string>('Kubeconfig');
     if (userKubeconfigPath) {
+      this.kubeconfigPath = userKubeconfigPath;
```

Comment:
Moving this here, or the `kubeconfigPath` value (used in `refresh`). could be wrong when a user-defined value is used and the files does not exist initially.